### DISCUSSION TITLE
Clipboard problem

### DISCUSSION THREAD

**Discussion by johdcmlaker2000** [tetrane/reven-community#7] [Q&A] [answered]
I noticed that after using Axion on Windows platform for a long time, the copy to clipboard option stops working. Restarting Axion does not help. Is this a known problem?

**Comment by dureuill** [comment]
Hi,

This bug is not known. Could you provide some information about your environment (eg, REVEN edition and version, way you installed REVEN - Linux Buster or Docker, Bare metal or VM, ...) so that we can attempt to reproduce it?

In particular, how are you using Axion?

1. In your browser (through VNC)
2. Using the [appimage](https://doc.tetrane.com/free/latest/Release-Notes/Packaging/AppImage.html#appimage-content).
   - if using the appimage, on which environment? If Windows, are you using WSL1 or WSL2? With which Linux distribution? 

Thank you for any additional information!

**Reply by johdcmlaker2000** [reply]
Hi, dureuill

i'm using REVEN Free edition, 2.10.1, host installed on Debian Buster Linux debian 4.19.0-18-amd64 #1 SMP Debian 4.19.208-1 (2021-09-29) x86_64 GNU/Linux

Axion is started through VNC, browser is the last Brave version 1.32.113 x64, based on Chromium: 96.0.4664.45

**Comment by johdcmlaker2000** [comment, marked as answer]
Also i noticed that the all data is not copied to the host system clipboard, but is displayed correctly in the VNC clipboard view widget. And from there it can be copied to host system clipboard

**Reply by dureuill** [reply]
Wow, this is new to me! AFAIK the VNC clipboard view widget was the only way to copy from the Axion GUI to the host clipboard. Looking further, there are [some issues](https://github.com/novnc/noVNC/issues?q=is%3Aissue+clipboard) on novnc indicating that copy/paste through VNC sometimes work, but is a known complicated problem.

To sidestep the VNC clipboard issues, you can either render [Axion natively in your REVEN VM](https://doc.tetrane.com/free/latest/Installation/Client/Local.html#enabling-local-rendering-of-vms-and-axion) or [install Axion natively in your host](https://doc.tetrane.com/free/latest/Installation/Client/Remote.html#installing-the-axion-appimage-on-a-remote-client).

**Reply by johdcmlaker2000** [reply]
> Wow, this is new to me! AFAIK the VNC clipboard view widget was the only way to copy from the Axion GUI to the host clipboard. Looking further, there are [some issues](https://github.com/novnc/noVNC/issues?q=is%3Aissue+clipboard) on novnc indicating that copy/paste through VNC sometimes work, but is a known complicated problem.
> 
> To sidestep the VNC clipboard issues, you can either render [Axion natively in your REVEN VM](https://doc.tetrane.com/free/latest/Installation/Client/Local.html#enabling-local-rendering-of-vms-and-axion) or [install Axion natively in your host](https://doc.tetrane.com/free/latest/Installation/Client/Remote.html#installing-the-axion-appimage-on-a-remote-client).

thanks for the answer

**Reply by dureuill** [reply]
Good, I'm marking this discussion as resolved then. Feel free to come back and continue the discussion if needed :slightly_smiling_face:
